test/metabase/models/data_permissions_test.clj

```diff
@@ -218,15 +223,43 @@
               (data-perms/set-table-permission! group-id-2 table-id-1 :perms/create-queries :query-builder)
               (is (= :query-builder (data-perms/table-permission-for-user user-id :perms/create-queries database-id table-id-1)))
               (data-perms/with-relevant-permissions-for-user user-id
-                ;; retrieve the cache now so it doesn't get counted in the call count
-                @data-perms/*permissions-for-user*
+                ;; Fetch the perms once to populate the cache
+                (data-perms/table-permission-for-user user-id :perms/create-queries database-id table-id-1)
                 ;; make the cache wrong
                 (data-perms/set-table-permission! group-id-1 table-id-1 :perms/create-queries :no)
                 ;; the cached value is used
                 (t2/with-call-count [call-count]
                   (is (= :query-builder (data-perms/table-permission-for-user user-id :perms/create-queries database-id table-id-1)))
                   (is (zero? (call-count))))))))))))
 
+(deftest inactive-table-permission-test
+  (testing "An inactive table appears as if it has no permissions, and is not cached"
```

Comment:
Awesome test, I was wondering about exactly this situation so it's great to see it explicitly laid out.

Reply:
Should we also have a test that queries fail? It's great that the permission is that it is blocked. But would like to see an actual query fail